### ISSUE TITLE
Get the difference between test set and validation set sorted

### ISSUE THREAD

**Issue by puria-radmard** [CamMARS2025/mars-steg#53]
@jskaf34 I don't see the point of having a val and test loader in `train_dataset_task.py`

In fact we don't even use the test loader in the script anywhere

Either we drop the test set, or we have some difference between the two (e.g. val set are examples where we know the model can get the answer correct)

**Comment by jskaf34** [comment]
For the train/valid/test split, we want to keep the test split to check if the model is really having the behaviour we want at the end of the training, it would be annoying if the model just learns by heart to replace for particular examples the equal signs for instance
the valid dataset is for the CoT evaluation Gap during training, having a random sampling of it is a first super easy implementation, we could definitely go for a better sampling
